R/calculate_within_batch_ari.R

```diff
@@ -48,54 +48,20 @@ calculate_within_batch_ari <- function(individual_sce_list,
   }
 
   # Check that `pc_name` is in merged SCE object
-  if (!pc_name %in% reducedDimNames(merged_sce)) {
-    stop("The provided `pc_name` cannot be found in the `merged_sce`.")
+  if (!any(pc_list %in% reducedDimNames(merged_sce))) {
+    stop("One or more of the PC names provided in `pc_list` cannot be found in the `merged_sce`.")
   }
 
-  # Pull out the PCs or analogous reduction from merged object
-  merged_pcs <- reducedDim(merged_sce, pc_name)
-
-  # Cluster merged pcs only one time
-  merged_sce <- merged_sce |>
-    cluster_sce(
-      pca_name = pc_name,
-      BLUSPARAM = bluster::NNGraphParam(cluster.fun = "louvain", type = "jaccard"),
-      cluster_column_name = "merged_clusters"
-    )
-
-  merged_clusters <- merged_sce$merged_clusters |>
-    purrr::set_names(merged_sce[[batch_column]])
-
-  # For every batch id, cluster and then calculate ARI for that batch
-  all_ari <- batch_ids |>
-    purrr::map_dbl(\(batch){
-      # Cluster pc matrix for specified batch
-      individual_clustering_result <- individual_sce_list[[batch]] |>
-        cluster_sce(
-          pca_name = "PCA",
-          BLUSPARAM = bluster::NNGraphParam(cluster.fun = "louvain", type = "jaccard"),
-          cluster_column_name = "individual_clusters"
-        )
-
-      # Extract clusters from merged clustering for batch
-      clusters_to_keep <- grep(batch, merged_sce[[batch_column]])
-      batch_merged_clusters <- merged_clusters[clusters_to_keep]
-
-      # Calculate ARI between pre-merged clustering and post-merged clustering for the given batch
-      ari <- bluster::pairwiseRand(individual_clustering_result$individual_clusters,
-        batch_merged_clusters,
-        mode = "index"
+  # Calculate within-batch ARI values across list of PCs
+  within_batch_ari_tibble <-
+    purrr::map_df(
+      pc_list,
+      ~ calculate_within_batch_ari_pcs(
+        merged_sce = merged_sce,
+        batch_column = batch_column,
+        pc_name = .
       )
-
-      return(ari)
-    })
-
-  # Create tibble with ARI and batch id
-  within_batch_ari_tibble <- tibble::tibble(
-    ari = all_ari,
-    batch_id = batch_ids,
-    pc_name = pc_name
-  )
+    )
```

Comment:
```suggestion
    ) |>
    dplyr::bind_rows()
```
The `purrr::map` will return a list of data frames, so you will need to combine those into a single dataframe.